plugins/modules/vmware_guest_disk.py

```diff
@@ -92,11 +92,22 @@
      - ' - C(scsi_controller) (integer): SCSI controller number. Valid value range from 0 to 3.'
      - '   Only 4 SCSI controllers are allowed per VM.'
      - '   Care should be taken while specifying C(scsi_controller) is 0 and C(unit_number) as 0 as this disk may contain OS.'
-     - ' - C(unit_number) (integer): Disk Unit Number. Valid value range from 0 to 15. Only 15 disks are allowed per SCSI Controller.'
+     - ' - C(unit_number) (integer): Disk Unit Number. C(controller_type) is added for other controller types except'
+     - '   SCSI, so the valid value has different range for controller types:'
+     - '   Valid value range from 0 to 15, except 7 for SCSI Controller.'
+     - '   Valid value range from 0 to 29 for SATA controller.'
+     - '   Valid value range from 0 to 14 for NVME controller.'
      - ' - C(scsi_type) (string): Type of SCSI controller. This value is required only for the first occurrence of SCSI Controller.'
      - '   This value is ignored, if SCSI Controller is already present or C(state) is C(absent).'
      - '   Valid values are C(buslogic), C(lsilogic), C(lsilogicsas) and C(paravirtual).'
      - '   C(paravirtual) is default value for this parameter.'
+     - ' - C(controller_type) (string): This parameter is added for managing disks attaching other types of controllers,'
```

Comment:
Could you please implement this as `suboptions`?

Reply:
Sure, thanks.